book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+
+**Integration Service** - The service that uses the pipeline, snapshot and environment controllers to watch for Component builds, trigger and manage the testing pipelines and create releases based on the testing outcome(s).
+
+**IntegrationTestScenario** - The custom resource that describes separate PipelineRuns that are supposed to be run by the snapshot controller. Any number of them can be set by the user and each of them includes information about the Tekton pipeline that will be executed - Tekton bundle, pipeline name and parameters, and the Environment details. The Integration Test Scenarios can also specify the contexts where they will be applied - this can be used to define when the scenario will run (Component or Composite stage - or both). If more than one Integration Test Scenario has the same context, their respective PipelineRuns will be executed in parallel.
+
+**Ephemeral Environment** - The ephemeral copy of an existing environment that’s created by the Integration service for use during a specific Integration pipeline run. After the environment is copied and marked as ephemeral, the Integration service creates a new SnapshotEnvironmentBinding linking the Snapshot under test with the newly created Environment. This triggers a deployment of the Snapshot on the new Environment, handled by the GitOps service. Finally, the Integration service links the credentials for the Environment to the Integration Tekton pipeline run and starts the test run.
+
+
+## Resources 
+Below are the list of CR’s that the integration service is responsible for interacting and performing operations:
+
+### CREATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| Snapshot | Post Component build PipelineRun completes.  | Used as the source input for the test gate pipeline |
+| PipelineRuns | Post Component build PipelineRun completes  | To test the Snapshot  |
+| Release | Post PipelineRun if there is a ReleaseLink for the Application, if the auto-release flag is set on the ReleaseLink | To signal to the Release Service to automatically push the Snapshot to the Production Environment |
+
+### READ
+
+| Custom Resources | When? | Why? |
+|---|---|---|
+| Application & Component | Before creating the Snapshots | To know which build.ImageSpecs to use in the Snapshot |
+| IntegrationTestScenario | Before creating the Component/Composite PipelineRun(s) | To get the information for the PipelineRun - which Tekton bundles and Environments will be used |
+| ReleaseLink | Before creating the Release | To signal the Release Service for next environment promotion (prod) |
+
+### UPDATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| SnapshotEnvironmentBinding | Post Test PipelineRun | To promote the Snapshot to the next environment (Stage/Stable) |
+| Snapshot | Post Test PipelineRun | To mark the snapshot validated so that it can be promoted to the next environment |
```

Comment:
Isn't the snapshot was immutable once created?

Reply:
The list of components and their digests is immutable, the status of the Snapshot CR is what's updated here.
Maybe we can clarify that by changing `To mark the snapshot validated` to `To update the Snapshot's status to validated`.